server/src/service-layer/controllers/StripeWebhook.ts

```diff
@@ -1,38 +1,66 @@
 import { Controller, SuccessResponse, Route, Post, Request } from "tsoa"
 import Stripe from "stripe"
+import UserDataService from "data-layer/services/UserDataService"
+import AuthService from "business-layer/services/AuthService"
+import {
+  CHECKOUT_TYPE_KEY,
+  CheckoutTypeValues
+} from "business-layer/utils/StripeProductMetadata"
 
 @Route("webhook")
 export class StripeWebhook extends Controller {
   @Post()
   @SuccessResponse(200, "Webhook post received")
   public async receiveWebhook(@Request() request: any): Promise<void> {
-    console.log("webhook received")
-
     const stripe = new Stripe(process.env.STRIPE_API_KEY)
-
+    // ensure security of the endpoint by constructing an event
+    let event: Stripe.Event
     try {
-      const event: Stripe.Event = stripe.webhooks.constructEvent(
+      event = stripe.webhooks.constructEvent(
         request.rawBody,
         request.headers["stripe-signature"],
-        // process.env.STRIPE_LOCAL // test local api secret
-        process.env.STRIPE_API_SECRET
+        process.env.STRIPE_LOCAL
+        // process.env.STRIPE_API_SECRET
       )
-      switch (event.type) {
-        case "payment_intent.succeeded":
-          console.log("payment_intent.succeeded")
-          break
-        case "payment_method.attached":
-          console.log("payment_method.attached")
-          break
-        default:
-          console.log(`Unhandled event type ${event.type}.`)
-      }
-
-      return this.setStatus(200) // set status to 200 as success
     } catch (err) {
       console.error(err)
-      // set status to 400 due to bad request
-      return this.setStatus(400)
+      return this.setStatus(401) // unauthorized request
+    }
+    // console.log(event.type)
+    if (event.type === "payment_intent.succeeded") {
+      // console.log("[WEBHOOK] received: payment_intent.succeeded")
+      console.log(request.body)
+      const userService = new UserDataService()
+      const authService = new AuthService()
+      // Stripe PaymentIntent
+      const { id } = event.data.object
+      // Fetch the checkout session from the PaymentIntent ID
+      const stripeSession: Stripe.Checkout.Session =
+        await stripe.checkout.sessions.retrieve(id)
```

Comment:
i believe we have a method for this in `StripeService` already, which was created in #237

Reply:
Whoops, yep just made the changes, has been a long time 😂